content/docs/work-with-source-git.md

```diff
@@ -0,0 +1,123 @@
+
+For downstream patches, packit creates a patch from git commits on top of
+`upstream_ref`. Downstream packaging changes are ignored in patch files since
+they can be directly committed to dist-git. You can control the ignore
+mechanism with config option `patch_generation_ignore_paths`.
+
+
+#### Example
+
+Let's have a look at source-git repo for pacemaker package:
+```
+$ git log --online
+bd12722 (HEAD -> c8s-test, origin/c8s-test, master) downstream packaging
+7201e28 Refactor: controller: remove unused argument
+91c557c Refactor: controller: convert active_op_t booleans to bitmask
+2e90063 Refactor: controller: rename struct recurring_op_s to active_op_t
+3ee90ce (tag: source-git-start) pacemaker-2.0.3 base
+```
+
+We have 5 commits:
+* 3ee90ce - the bottom commit, equals to an unpackaed upstream tarball
+* 2e90063 - a downstream patch
+* 7201e28 - a downstream patch
+* 91c557c - a downstream patch
+* bd12722 - top commit, contains a spec file so packit won't create a patch file out of this one
+
+packit.yaml has only two lines:
+```
+$ cat .packit.yaml
+specfile_path: SPECS/pacemaker.spec
+upstream_ref: source-git-start
+```
+
+As you can see, we are telling packit, that commit tagged as `source-git-start`
+indicates the barrier between upstream and downstream, in our case it's just a
+single commmit with the upstream archive content, but it could have been the
+whole upstream git history.
+
+
+### Adding changes
+
+So, how can one add new changes into a source-git repo? That's simple - just
+commit them. If your repo is on GitHub (or another public forge), you can even
+force-push changes to achieve desired results with patches.
+
+
+### Creating a SRPM
+
+The format between a source-git repo and the build system is a source RPM,
+SRPM. We can then take the file and build it locally, send it to koji or copr
+and we would get the result we want - a list of binary RPMs created from the
+SRPM.
+
+In the end, you can create SRPM with packit's `srpm` command:
+```
+$ packit srpm
+SRPM: /home/tt/p/c/s/pacemaker/pacemaker-2.0.3-6.gbd127227.fc32.src.rpm
+```
+
+#### How packit generates a SRPM from a source-git repo?
+
+These are the steps:
+
+1. Download archive specified in specfile's Source directive.
+2. Process commits on top off `upstream_ref`.
+3. Bump release in the spec file.
+4. Generate new changelog entry in the spec file.
+5. Run rpmbuild and set paths so that rpmbuild can find patches, spec files,
+   archive and additional sources.
+
+
+### Updating your package in Fedora
+
+When you use source-git to track a package in Fedora, the workflow is the same as if you were in an upstream repository:
+```
+$ packit propose-update
+```
+
+and that results in this PR: [src.fedoraproject.org/rpms/python-docker/pull-request/26](https://src.fedoraproject.org/rpms/python-docker/pull-request/26).
+
+
+If you are looking for the design concept of source-git, please head on here: [source-git/design]({{ ref "/docs/source-git/design" }})
```

Comment:
Having this sentence as the second sentence in this document would make more sense I think.